tests/test_multiprocessing.py

```diff
@@ -0,0 +1,130 @@
+import pytest
+import multiprocessing
+import contextlib
+
+from redis.connection import Connection, ConnectionPool
+from redis.exceptions import ConnectionError
+
+
+@contextlib.contextmanager
+def exit_callback(callback, *args):
+    try:
+        yield
+    finally:
+        callback(*args)
+
+
+class TestMultiprocessing(object):
+    # Test connection sharing between forks.
+    # See issue #1085 for details.
+
+    def test_connection(self):
+        conn = Connection()
+        assert conn.send_command('ping') is None
+        assert conn.read_response() == b'PONG'
+
+        def target(conn):
+            assert conn.send_command('ping') is None
+            assert conn.read_response() == b'PONG'
+            conn.disconnect()
+
+        proc = multiprocessing.Process(target=target, args=(conn,))
+        proc.start()
+        proc.join(3)
+        assert proc.exitcode is 0
+
+        # Check that connection is still alive after fork process has exited.
+        with pytest.raises(ConnectionError):
```

Comment:
The comment on the previous line says the connection should still be alive after the fork has exited. Why are we expecting a `ConnectionError` here then?

If `ConnectionError` is in fact expected, let's change the comment to indicate such and remove the `assert` statements within the `pytest.raises` block. Just looking at this I don't know if we're actually testing that `conn.send_command('ping') is None` and `conn.read_response() == b'PONG'` or if those lines are just there to trigger the `ConnectionError`. If they're just triggering the `ConnectionError`, there's no need to assert on them.